plugins/gatsby-plugin-generate-doc-json/utils/handlers.js

```diff
@@ -13,23 +20,37 @@ const getAllAttributes = (node) =>
       }, {})
     : {};
 
+const getSrcUrl = (fileRelativePath, url) =>
+  path.join(
+    path.dirname(fileRelativePath.replace('src/content', '')),
+    url.replace('./', '')
+  );
+
 module.exports = {
   image: (h, node, imageHashMap, fileRelativePath) => {
-    const srcUrl = fileRelativePath
-      .replace('.mdx', '')
-      .replace('src/content/', '')
-      .split('/')
-      .slice(0, -1)
-      .join('/')
-      .concat(node.url.replace('./', '/'));
+    const srcUrl = getSrcUrl(fileRelativePath, node.url);
+
+    const isBlockImage =
+      isMdxElement('paragraph', node.parent) &&
+      node.parent.children.length === 1;
+
     return h(
       node,
-      'img',
+      isBlockImage ? 'div' : 'span',
       {
-        src: imageHashMap[srcUrl] || node.url,
-        alt: node.alt,
+        className: [isBlockImage ? 'block-image' : 'inline-image', 'image'],
```

Comment:
I see that you're setting `data.style` in the `ImageSizing` handler (which is 💯 ), but you'll also need to access it here and set it as a property, otherwise the data just gets discarded.

One other question: Does it makes sense to do the style string compilation here or from the parent component that wants to add inline styles to the image? That would affect where you call `compileStyleString`.

One advantage of doing it here vs a parent is that the parent would then be able to use object literal syntax to set styles on `data.style` rather than having to convert to a string. Thoughts? Totally fine the way it is, but figured I'd spur some discussion 🙂

Reply:
ah I set it on the image, but I'm going to want to pass it in as 
```js 
stripNulls({
        className: [isBlockImage ? 'block-image' : 'inline-image', 'image'],
        style: get(node, 'data.style', null),
```

correct?

Also: when you say object literal syntax, what do you exactly mean?

Reply:
I just mean this:

```js
const style = {
  display: 'inline-block',
  margin: '0 auto'
}
```

vs this:

```js
const style = 'display: inline-block; margin: 0 auto;
```

Typically "object literal" refers the the bare object as a variable (i.e. `{ }`). Since everything is technically an "Object" in JavaScript, the term "object literal" was meant to describe those kinds of variables. Does that help?

Reply:
And yes! That change looks great 😄